### DISCUSSION TITLE
A question about the locale

### DISCUSSION THREAD

**Discussion by SoumyaK4** [cotes2020/jekyll-theme-chirpy#2115] [Q&A] [answered]
I am unable to test my site from different locales, in all the locales I've tried with [locabrowser](https://www.locabrowser.com/)
It's showing content in English.

The _config has  `lang : ` not set to anything right now, before it was set to en, but in both cases I'm getting all results in en.

Can someone tell me how I can make my site shown in a different language please 

Thank you 🙏🏻

**Comment by SoumyaK4** [comment]
Just to be clear 
I know that setting the lang in config sets that as main language, or I can specify the language at post frontmatter...
But I'm just looking for a way to test how people see ny site in other languages....
And according to goatcounter people have read my content with Korean and Russian, but I can't figure out any way of testing that myself...
Sorry for my limited knowledge 😔

**Comment by tis-starlight** [comment]
It seems that **GoatCounter** might be showing the language based on the browser's language setting, not necessarily indicating that the user is viewing your site in Korean or Russian.

You could try copying the specific language **locale** from `jekyll-theme-chirpy` into your site's directory, and then enable that language in your `_config.yml` file. This might help align the language display with your site content.

**Reply by SoumyaK4** [reply]
doing that works as I mentioned above,
but I want to check if my site is dynamically serving different locales

**Reply by tis-starlight** [reply]
I have no idea about "dynamically serving different locales", but from what I've seen, you translate the content.

**Reply by SoumyaK4** [reply]
oh I guess I had the wrong impression that people with ip at korea and browser language set to korean would get a korean version from the _data/locale.

I'll still keep this thread open for a few more days in hope that someone can give me a dynamic solution

**Comment by kungfux** [comment, marked as answer]
Hmm, thanks for asking @SoumyaK4. I thought it works dynamically, but I never got a chance to see it in action.

You can use Chrome DevTools / Sensors / Location to test site behavior based on locale.

Here are my observations.

The config says:

> If it has the same name as one of the files in folder `_data/locales`, the layout language will also be changed,
> otherwise, the layout language will use the default value of 'en'.
> lang: en

So, if `lang` is not set, then `en` should be used by default.

If you build a website locally and check for example `_site/index.html`, you will find that text values are already set based on `lang`. Also, there are no javascript files that use files from the `locale` folder. So there is no dynamic language handling as far as I can see.

So if you really need it, you have to implement it from scratch.

Personally, I'm glad it's static so I can use English only everywhere, including posts. It will be a huge overhead to support multiple languages. This is too much for one person doing this as a hobby.

**Reply by SoumyaK4** [reply]
Thanks
Glad to know it's not dynamic as I want it to be only English.
Though now that I think about it, I'm stupid, Jekyll is ssg, how can it be dynamic 🙆🏻
